src/main/java/org/pesmypetcare/webservice/dao/MedicationDaoImpl.java

```diff
@@ -0,0 +1,185 @@
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        List<Map<List<String>, Object>> externalList = new ArrayList<>();
+        String aux;
+        ApiFuture<QuerySnapshot> future = medicationsRef.get();
+        List<QueryDocumentSnapshot> medicationDocuments = null;
+        try {
+            medicationDocuments = future.get().getDocuments();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        } catch (ExecutionException e) {
+            e.printStackTrace();
+        }
+        for (QueryDocumentSnapshot medicationDocument : medicationDocuments) {
+            Map<List<String>, Object> internalList = new HashMap<>();
+            List<String> pks = new ArrayList<String>();
+            aux = medicationDocument.getId();
+            if (initialDate.compareTo(pkToDate(aux)) < 0 && finalDate.compareTo(pkToDate(aux)) > 0){
+                pks.add(pkToDate(aux));
+                pks.add(pkToName(aux));
+                internalList.put(Collections.singletonList("dateName"), pks);
+                internalList.put(Collections.singletonList("body"),
+                        medicationDocument.toObject(MedicationEntity.class));
+                externalList.add(internalList);
+            }
+        }
+        return externalList;
+    }
+
+    @Override
+    public Object getMedicationField(String owner, String petName, String dateName, String field)
+            throws DatabaseAccessException {
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        DocumentReference medicationDocRef = medicationsRef.document(dateName);
+        ApiFuture<DocumentSnapshot> future = medicationDocRef.get();
+        DocumentSnapshot medicationDoc;
+        try {
+            medicationDoc = future.get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+        if (!medicationDoc.exists()) {
+            throw new DatabaseAccessException(INVALID_MEDICATION_EXC, MEDICATION_DOES_NOT_EXIST_EXC);
+        }
+        return medicationDoc.get(field);
+    }
+
+    @Override
+    public void updateMedicationField(String owner, String petName, String dateName, String field,
+                                      Object value) {
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        DocumentReference medRef = medicationsRef.document(dateName);
+        medRef.update(field, value);
+    }
+
+
+    public CollectionReference getMedicationsRef(String owner, String petName) {
+        return db.collection("users").document(owner).collection("pets").document(petName).collection("medications");
+    }
+
+    public String toPK(String date, String name){
+        return date + "%" + name;
+    }
+
+    public String pkToDate(String pk){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['{' is not preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)